### PR TITLE
Add Cluster Credentials Data Source to allow using password and certificate download in Terraform config

### PR DESCRIPTION
**Pull request by hail-insta**
This PR adds a new data source that retrieves an existing cluster's password and certificate download link.
Addressing [this](https://github.com/instaclustr/terraform-provider-instaclustr/issues/24) issue

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 BIN_NAME="terraform-provider-instaclustr"
-VERSION=v1.6.1
+VERSION=v1.7.1
```
Comment (arjunrajshekhar) [inline Makefile:2]: The next version would be v1.7.0 😄

File: test/resource_cluster_test.go [hunk 2]
```diff
@@ -119,29 +119,6 @@ func TestKafkaConnectClusterCreateNonInstaclustrAZURE(t *testing.T) {
 	})
 }
 
-func TestKafkaConnectClusterInvalid(t *testing.T) {
```
Comment (hail-insta) [inline test/resource_cluster_test.go:122]: Removed because `data/invalid_kafka_connect.tf` does not exist.

File: README.md [hunk 3]
```diff
@@ -96,10 +96,14 @@ resource "instaclustr_cluster" "example" {
 
 ## Configuration
 ### Resources
-### Resource:  `instaclustr_cluster`  
-A resource for managing clusters on Instaclustr Managed Platform. A cluster contains a base application and several add-ons.
+### Resource:  `instaclustr_cluster` and Data Source `instaclustr_cluster_credentials`
```
Comment (arjunrajshekhar) [inline README.md:99]: Could the data source be documented under a separate heading just below `instaclustr_cluster`?

File: instaclustr/api_client.go [hunk 4]
```diff
@@ -297,6 +297,27 @@ func (c *APIClient) DeleteEncryptionKey(keyID string) error {
 	return nil
 }
 
+func (c *APIClient) ReadClusterCredentials(clusterID string) (*ClusterCredentials, error) {
```
Comment (arjunrajshekhar) [inline instaclustr/api_client.go:300]: [Desired, reusability] - could `ReadCluster` be used instead? Accordingly you could just enhance the `Cluster` struct

File: Makefile (hunk 1 above)
Comment (hail-insta) [reply Makefile]: I have updated to make it v1.7.0

File: README.md (hunk 3 above)
Comment (hail-insta) [reply README.md]: I have updated to document the new data source under a separate heading below `instaclustr_cluster`

File: instaclustr/api_client.go (hunk 4 above)
Comment (hail-insta) [reply instaclustr/api_client.go]: I have updated to reuse the existing `ReadCluster` and `Cluster` struct

**Review by arjunrajshekhar — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -167,6 +167,15 @@ dedicated_zookeeper|Indicate whether this Kafka cluster should allocate dedicate
...
 zookeeper_node_count|If `dedicated_zookeeper` is true, then it indicates how many nodes are allocated to be Zookeeper nodes|Kafka
 
+### Data Source `instaclustr_cluster_credentials`
+A read-only data source used to get the password and certificate download link of a cluster.
+
+Property | Description | Default
+---------|-------------|--------
+cluster_id|The ID of an existing Instaclustr cluster.|Required
+cluster_password|The password of the existing Instaclustr cluster.|Computed
+certificate_download|The certifiacte download link of the existing Instaclustr cluster.|Computed
```
Comment (h3nd24) [inline README.md:177]: minor typo here: certifiacte -> certificate

**Review by h3nd24 — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (hail-insta) [reply README.md]: I have fixed the typo

**Review by arjunrajshekhar — APPROVED** [review]

**Review by h3nd24 — APPROVED** [review]

**Pull request merged**